### PR TITLE
Adding the Open Fuel Cycle Benchmark folder

### PR DESCRIPTION
**Pull request by Baaaaam**
Adding a dedicated folder for the different tests of the Open Fuel Cycle Benchmark (OFCB):
it contains a folder for each test:
* test1:
     * the json file: canonical definition of the test 1
     * CYCLUS : folder containing the cyclus input for test 1 separated in 2 cases: batched and unbatched mod.
     * a dedicated folder for the recipe for the UOX fuel (3.25% U5 enrich in a 900 GWe LWR reactor) loaded taken from the CLASS Database (including a file describing the format used)
* test2:
   * empty json file

### CODE REVIEW

File: input/OFCB/test1/CYCLUSinput/UnBatched/1.xml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		<nuclide><id>952420000</id>	<comp>0.0040</comp></nuclide>
+		<nuclide><id>80160000</id>	<comp>200</comp></nuclide>
+	</recipe>
+
+</simulation>
+
+
+
+
+
```
Comment (scopatz) [inline input/OFCB/test1/CYCLUSinput/UnBatched/1.xml]: What's with all the whitespace? Just curious....

File: input/OFCB/test1/STD900.info [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Reactor REP
+Fueltype MOX
```
Comment (scopatz) [inline input/OFCB/test1/STD900.info]: What is an "info" file? This isn't in a standard format...

File: input/OFCB/test1/pb1.json [hunk 3]
```diff
@@ -0,0 +1,93 @@
+{
+	"materials": {
+		"pwr_uox": {"attributes": {"recipe": true}, 
+			"constraints": [
```
Comment (scopatz) [inline input/OFCB/test1/pb1.json]: Are these TABs?

File: input/OFCB/test1/STD900.log [hunk 4]
```diff
@@ -0,0 +1,23 @@
+
+===================================================
+---------------------------------------------------
+
```
Comment (scopatz) [inline input/OFCB/test1/STD900.log]: What is the purpose of having a log file checked in?

File: input/OFCB/test1/STD900.info (hunk 2 above)
Comment (Baaaaam) [inline input/OFCB/test1/STD900.info]: this is the basic CLASS file coming with a recipe base... it contains some information require by CLASS

**Comment by scopatz** [comment]
The dat file seems to be some custom ascii format. Any chance of converting to something more standard or a description of the format? Thanks @Baaaaam

File: input/OFCB/test1/pb1.json (hunk 3 above)
Comment (Baaaaam) [inline input/OFCB/test1/pb1.json]: yes.
Should I use space ?

File: input/OFCB/test1/STD900.log (hunk 4 above)
Comment (Baaaaam) [inline input/OFCB/test1/STD900.log]: correct, I put all file coming the the base in CLASS.. the log is not needed...

File: input/OFCB/test1/CYCLUSinput/UnBatched/1.xml (hunk 1 above)
Comment (Baaaaam) [inline input/OFCB/test1/CYCLUSinput/UnBatched/1.xml]: no use

**Comment by Baaaaam** [comment]
@scopatz have you a format in mind ? 
I could easily add a description of the file format, but I don't know any "standart" format to decribe fuel evolution...

File: input/OFCB/test1/pb1.json (hunk 3 above)
Comment (scopatz) [inline input/OFCB/test1/pb1.json]: Yes. Please remove the tab key from your keyboard whenever you open a text editor. Kindly replace it when you open a shell :)

File: input/OFCB/test1/CYCLUSinput/UnBatched/1.xml (hunk 1 above)
Comment (scopatz) [inline input/OFCB/test1/CYCLUSinput/UnBatched/1.xml]: Can you remove it then, please?

File: input/OFCB/test1/CYCLUSinput/UnBatched/1.xml [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+			<feed_commod>natl_u</feed_commod>
+			<product_commod>enriched_u</product_commod>
+			<tails_commod>ef_tails</tails_commod>
+			<feed_recipe>natl_u</feed_recipe>
+			<tails_assay>0.003</tails_assay>
+			<max_feed_inventory>500</max_feed_inventory>
+			</Enrichment>
+		</config>
+		</facility>
+	-->
```
Comment (scopatz) [inline input/OFCB/test1/CYCLUSinput/UnBatched/1.xml]: Remove extraneous comment.

File: input/OFCB/test1/pb1.json [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+			"constraints": [],
+			"inputs": ["spent_pwr_uox"]
+		}
+	},
+	
+	"fuelCycle": {
+		"attributes": {
+			"grid": "year",
+			"initialConditions": {
+				"lwr_reactor_power": 1
```
Comment (scopatz) [inline input/OFCB/test1/pb1.json]: this line needs a trailing comma

File: input/OFCB/test1/pb1.json (hunk 3 above)
Comment (katyhuff) [inline input/OFCB/test1/pb1.json]: @Baaaaam :
Depending on your text editor, you can usually set a preference to never use tab. If you use vi, you can make this happen by adding the following line to your ~/.vimrc file:

:set expandtab

**Comment by scopatz** [comment]
> @scopatz have you a format in mind ?
I could easily add a description of the file format, but I don't know any "standart" format to decribe fuel evolution...

Unless I am missing something, there is no description, link, reference, citation, or documentation in this PR or repo for the format and meaning of the dat file.  We need to be able to point to a specification somewhere for this to be useful.

File: input/OFCB/test1/pb1.json (hunk 3 above)
Comment (Baaaaam) [inline input/OFCB/test1/pb1.json]: should be ok now.

Comment (scopatz) [inline input/OFCB/test1/pb1.json]: Cool thanks.  Can you now reduce to a normal number of space.  For JSON it is traditional to use 1, 2, or 4 spaces for indentation.  Thanks!

**Comment by Baaaaam** [comment]
@scopatz  I have added a README file in the fuel recipe folder which contain the description of the format used.

Sorry for the different mistakes I did/will do, it my first pull request... if there is thing I still do wrong please tell me...

File: input/OFCB/test1/pb1.json (hunk 3 above)
Comment (Baaaaam) [inline input/OFCB/test1/pb1.json]: should be correct now with a 2 spaces indentation

File: input/OFCB/test1/fuelrecipe/STD900.info [hunk 7]
```diff
@@ -0,0 +1,9 @@
+Reactor REP
+Fueltype MOX
+CycleTime 4.99795
```
Comment (scopatz) [inline input/OFCB/test1/fuelrecipe/STD900.info]: What are the units here?

File: input/OFCB/test1/fuelrecipe/STD900.info [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+AssemblyHeavyMetalMass 45168.6 g
+ConstantPower 1.77389e+06 W
+CutOff 1e-10
```
Comment (scopatz) [inline input/OFCB/test1/fuelrecipe/STD900.info]: Units? CutOff for what?

**Comment by scopatz** [comment]
Hi @Baaaaam, the info files still don't seem to have sufficient documentation.

File: input/OFCB/test1/fuelrecipe/README.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+----------------------------
+
+the ".dat" file is formated in line : each line contain the evolution of a parameter at each timestep. A keyword is placed in the begining of each line to be able to identify the parameter described.
+
+List of keyword used :
+
+ * **time**: time in s of each step
+ * **keff**: neutron multiplication factor 
+ * **flux**: neutron flux in neutron/(second.cm2)
+ * **inv**: followed by Z A I, evolution of the quantity of the nucleus Z A I in atoms
```
Comment (scopatz) [inline input/OFCB/test1/fuelrecipe/README.md]: What is meant by evolution. Is this an array of the length of the time step?

**Comment by Baaaaam** [comment]
Since it is difficult to clearly define a canonical way to describe a fuel evolution in a canonical way and the fuel evolution is not a real issue there ( test 1 is suppose to test the fuel loading and discharge of a reactor), I have decided to remove the fuel recipe... It is not a real solution, and just delay the time we will actually face it...

The require information for the fuel are provided in the pb1.jsn file and the README file as a recipe for the fuel loading and discharge...

**Comment by Baaaaam** [comment]
@gonuke @scopatz @rwcarlsen, I have completed all the requested change... Could you take a look and notify me if there is still some change/adjustment to make ?
I would like to use this as a support for the open fuel cycle benchmark invitation.

Thx
BaM

**Comment by scopatz** [comment]
This looks good @Baaaaam, thanks and sorry about the ANS-induced delay.

**Comment by Baaaaam** [comment]
no pb!
Thank you for your comment!

**Pull request merged**